save-backend/src/main/kotlin/org/cqfn/save/backend/security/OrganizationPermissionEvaluator.kt

```diff
@@ -0,0 +1,71 @@
+package org.cqfn.save.backend.security
+
+import org.cqfn.save.backend.service.LnkUserOrganizationService
+import org.cqfn.save.backend.utils.AuthenticationDetails
+import org.cqfn.save.domain.Role
+import org.cqfn.save.entities.Organization
+import org.cqfn.save.permission.Permission
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+
+/**
+ * Class that is capable of assessing user's permissions regarding organizations.
+ */
+@Component
+class OrganizationPermissionEvaluator {
```

Comment:
Some logic looks quite similar to the `ProjectPermissionEvaluator`, could you please look, can we extract common parts into the separate module? Probably we can accept in these methods some lambdas as last arguments, which will contain     individual logic for project or organization, but the most part of permissions evaluation will be the same?

Reply:
Well there are some common parts but they are kinda small. The point is that `ProjectPermissionEvaluator` is bigger and has more cases to check. To my mind it is better to split this logic

Reply:
There are some common parts of code, but to my mind they are too small to move these two permission evaluators to the same class. To my mind it is better to keep the logic split.

Reply:
Discussed with @petertrr , decided to leave as is because in the nearest future this classes will differ more.